packages/tenant-process/src/services/tenantService.ts

```diff
@@ -938,28 +965,37 @@ export function tenantServiceBuilder(
       );
       return readModelService.getTenantsByName({ name, offset, limit });
     },
-    async getTenantById(
-      id: TenantId,
-      logger: Logger
-    ): Promise<WithMetadata<Tenant> | undefined> {
+    async getTenantById(id: TenantId, logger: Logger): Promise<Tenant> {
       logger.info(`Retrieving tenant ${id}`);
-      return readModelService.getTenantById(id);
+      const tenant = await retrieveTenant(id, readModelService);
```

Comment:
The `not found` error was being raised in the router. Now this happens inside `retrieveTenant`.